### PR TITLE
AF-2799: Remove mirrors from tests

### PR DESCRIPTION
**Pull request by corwinsheahan-wf**
## Ultimate problem:
Usages of dart mirrors go away in dart 2

## How it was fixed:
Remove usages of dart mirrors. 

## Testing suggestions:
[ ] CI Passes

## Potential areas of regression:
Test coverage

---

> __FYA:__ @greglittlefield-wf @aaronlademann-wf @kealjones-wk @evanweible-wf @sebastianmalysa-wf

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/test_util/prop_utils.dart [hunk 1]
```diff
@@ -14,35 +14,13 @@
...
-
-void testKeys(List<String> keys, dynamic instanceBuilder()) {
+void testInvalidKey(dynamic instanceBuilder()) {
   test('cannot set / read values that are not its prop map', () {
     var instance = instanceBuilder();
     expect(() {instance['notThere'];},
         throwsA(hasToStringValue(contains('Map does not contain this key'))));
   });
-  for (var propKey in keys) {
-    test('prop: $propKey can have its value set / read', () {
```
Comment (corwinsheahan-wf) [inline test/test_util/prop_utils.dart]: These tests were only intended to verify that the code generation for implementing concrete getters/setters was working properly, and that these props classes had a prop key namespace of `''`, (except for the case of aria props, which are prefixed with `aria-`). Therefore, these tests have been simplified [here](https://github.com/Workiva/over_react/pull/219/files#diff-33b0e4fc8ff490302629b793a0245148R26) to just test on a few of the props on a mixin.

File: test/test_util/mock_classes.dart [hunk 2]
```diff
@@ -17,29 +17,6 @@ library mock_classes;
...
-  'MockFileUploadInputElement',
-  'dart.dom.html.FileList',
-  'dart.dom.html.File',
-  'dart.dom.html.FileUploadInputElement',
-  'MockSyntheticEvent',
-  'MockSyntheticMouseEvent',
-  'react.SyntheticEvent',
-  'react.SyntheticMouseEvent',
-], override: 'mockito')
-import 'dart:mirrors';
```
Comment (corwinsheahan-wf) [inline test/test_util/mock_classes.dart]: @greglittlefield-wf were these needed? afaik, the only usage of mirrors within mockito was `spy()`, which I didn't find any usages of in `over_react`'s tests.

**Comment by robbecker-wf** [comment]
+10 CI does indeed pass

File: test/over_react/component/dom_components_test.dart [hunk 3]
```diff
@@ -83,12 +82,12 @@ main() {
       if (expectedTagName == 'missingGlyph') expectedTagName = 'missing-glyph';
       if (expectedTagName.startsWith(new RegExp('svg.'))) expectedTagName = expectedTagName.substring(3);
 
-      test('Dom.$name generates the correct type', () {
-        DomProps builder = domClassMirror.invoke(element, []).reflectee;
+      test('${method.toString()} generates the correct type', () {
```
Comment (evanweible-wf) [inline test/over_react/component/dom_components_test.dart]: Using `.toString()` inside a string interpolation is redundant:
```suggestion
      test('$method generates the correct type', () {
```

File: test/test_util/mock_classes.dart (hunk 2 above)
Comment (greglittlefield-wf) [reply test/test_util/mock_classes.dart]: Mm, looks like these were added under the false assumption that `Mock` classes used mirrors.

Away with them! 😄

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-pp

File: test/over_react/component/dom_components_test.dart (hunk 3 above)
Comment (corwinsheahan-wf) [reply test/over_react/component/dom_components_test.dart]: 🤦‍♂️

**Pull request merged**